scripts/tsvFindCompositions.js

```diff
@@ -0,0 +1,166 @@
+    composingIds.forEach( (anId, index, theArray ) => {
+      theArray[index] = parseInt(anId);
+    });
+    if (composingIds.length > 0 && !checkCompRepeats(aComposition.id, composingIds)) {
+      aComposition.composingIds = composingIds;
+    }
+    compositions.push(aComposition);
+  }
+  return compositions;
+};
+
+/**
+ * Check the composition array for the given symbol.  If the composition is only
+ * a repetition of the symbol itself, ignore it.
+ * @param {Integer} symbolID - BCI AV ID of the symbol in question
+ * @param {Array} compositionIDs - Set of BCI AV IDs that the `symbolID` symbol
+ *                                 is composed out of.
+ * @return {Boolean} - `true` if the `compositionIDs` have one member whose
+ *                      value is the same as the `symbolID`; otherwise, false.
+ */
+function checkCompRepeats (symbolID, compositionIDs) {
+  let result = false;
+  if (compositionIDs && compositionIDs.length === 1) {
+    if (compositionIDs[0] === symbolID) {
+      result = true;
+    }
+  }
+  return result;
+}
+
+/**
+ * Merge the composition information into the main Bliss symbol json file
+ * and save to a new file
+ * @param {Array} compositions - Array of objects whose `character` and
+ *                              `composition` fields are to be copied into
+ *                              the main Bliss symbols array of objects
+ * @param {Array} mainBlissSymbols - The main array of Bliss symbols information
+ * @param {String} filePath    - The path to the file to store the merged JSON
+ *                               data to.
+ */
+function mergeAndSave(compositions, mainBlissSymbols, filePath) {
+  // Merge
+  compositions.forEach((aComposition) => {
+    const blissRecord = mainBlissSymbols.find((symbol) => parseInt(symbol.id) === aComposition.id);
+    if (!blissRecord) {
+      console.error(`Not found: ${aComposition.id}:${aComposition.gloss}`);
+    }
+    else {
+      blissRecord.isCharacter = aComposition.isCharacter;
+      if (aComposition.composingIds) {
+        blissRecord.composingIds = aComposition.composingIds;
+      }
+    }
+  });
+  // Save to disk
+  saveJsonToFile(filePath, mainBlissSymbols);
+}
+
+/**
+ * Store the json data to the given file
+ * @param {String} filePath - Path to the file to store the fiven json
+ *                            information
+ * @param {Array} jsonData - The JSON data to write to a disk file
+ */
+async function saveJsonToFile (filePath, jsonData) {
+  try {
+    await writeFile(filePath, JSON.stringify(jsonData, null, 2));
+  }
+  catch (err) {
+    console.error("Failed to save to '`${filePath}`'");
+    console.error(err);
+  }
+}
+
+/**
+ * Merge the composition data with the main `bliss_symbol_explanations.json`
+ */
+async function main () {
+  const compositions = await findCompositions();
+  await saveJsonToFile("./data/blissSymbolComposition.json", compositions);
```

Comment:
It seems "blissSymbolComposition.json" is a subset of "bliss_symbol_explanations.json", is it necessary to save it? When the code description says two files are generated but I found "blissSymbolComposition.json" is not saved in the "/data" directory, I wondered why it's not saved until I noticed it's a subset of "bliss_symbol_explanations.json". Please let me know if I missed anything. Thanks.

Reply:
It's odd that you didn't get a `./data/blissSymbolComposition.json` file.  You should have.

But, you are correct that it contains only the new composition and character status information.  It was saved at one point for debugging, and I thought there was a small chance that it might remain useful in a standalone file since it is smaller than the full `bliss_symbol_explanations.json`.  That last operations the script does is to merge the new composition and character status with the existing `bliss_symbol_explanations.json` file and then save the merged file back out to `bliss_symbol_explanations.json`.  So, perhaps the `blissSymbolComposition.json` is no longer needed.  I have removed the code that saves it to disk.